packages/flutter/lib/src/widgets/drag_target.dart

```diff
@@ -612,12 +618,13 @@ class DragTarget<T extends Object> extends StatefulWidget {
     super.key,
     required this.builder,
     this.onWillAccept,
+    this.onWillAcceptWithDetails,
     this.onAccept,
     this.onAcceptWithDetails,
```

Comment:
Is this similar to these two? if so, why do we not assert in this case that both are not set?

Reply:
In the case of `onWillAccept` and `onWillAcceptWithDetails` there is a discrepancy, what if one returns `false` and the other returns `true`? This is not the case between `onAccept` and `onAcceptWithDetails` though since it would just trigger these calls and the user may very well implement two different functionalities in both callbacks, even though it does not make any sense in doing that. The question as to whether we "should" put an assert for these two as well is debatable.

Reply:
I think we should assert and make a plan to deprecate and migrate folks over to the newer APIs.

Reply:
> I think we should assert and make a plan to deprecate and migrate folks over to the newer APIs.

Should this new assertion for `onWillAccept` and `onWillAcceptWithDetails` be managed in a separate PR and issue?

Reply:
Yeah, let's do that in a separate PR. I can if you'd like with the deprecations and dart fixes to migrate folks to the newer APIs. I think this change is ok to go ahead with.